### PR TITLE
Improve MQTT examples to reflect better to new naming standards

### LINKED ISSUE #28427: Update HASS MQTT Discovery documentation for 2023.8 changes
### Feedback

Hi,
With the 2023.8 changes to the [Entity Naming Standards](https://developers.home-assistant.io/docs/core/entity/#entity-naming) and recent [dev blog post on MQTT naming](https://developers.home-assistant.io/blog/2023-057-21-change-naming-mqtt-entities/), it would help developers a if the [MQTT Discovery documentation](https://www.home-assistant.io/integrations/mqtt#mqtt-discovery) were updated to include these new resources, and the (now defunct) example test code updated.

1. Add a link to the recent [dev blog post on MQTT naming](https://developers.home-assistant.io/blog/2023-057-21-change-naming-mqtt-entities/)
1. Add a link to the [Entity Naming Standards](https://developers.home-assistant.io/docs/core/entity/#entity-naming), and how this impacts MQTT Discovery.
1. Update the `mosquitto_pub`j examples as they currently include are one character off triggering the discovered entities Repair message (`Garden` and `garden` which were valid, but now are confusing).

A possible updated developer MQTT Discovery test example is:
```
# Send a MQTT Discovery message to create a motion sensor
# Note the difference between the DEVICE name and the motion ENTITY (of which there could be several)
# Note the motion ENTITY name is not required, or can be set to`None` - see the HASS Developer Blog
mosquitto_pub -r -h 127.0.0.1 -p 1883 -t "homeassistant/binary_sensor/garden/config" -m '{"name": "Motion ENTITY", "device_class": "motion", "state_topic": "homeassistant/binary_sensor/garden/state", "unique_id": "motion01ad", "device": {"identifiers": ["01ad"], "name": "Garden Sensor DEVICE" }}'
```
A screenshot of the result which you can freely use is in this post:
https://community.home-assistant.io/t/psa-mqtt-name-changes-in-2023-8/598099/61

All the best,

James

### URL

https://www.home-assistant.io/integrations/mqtt/

### Version

2023.8.0

### Additional information

This docs update started from this post:
https://community.home-assistant.io/t/psa-mqtt-name-changes-in-2023-8/598099/61

### PR DESCRIPTION
**Pull request by jbouwh**
## Proposed change

Improve some examples to reflect better on new naming standards.

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #28427

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by jbouwh** [comment]
@FloatingBoater feel free to leave a review

File: source/_integrations/mqtt.markdown [hunk 1]
```diff
@@ -596,15 +596,19 @@ A motion detection device which can be represented by a [binary sensor](/integra
...
 
-It is also a good idea to add a `unique_id` to allow changes to the entity and a `device` mapping so we can group all sensors of a device together.
+It is also a good idea to add a `unique_id` to allow changes to the entity and a `device` mapping so we can group all sensors of a device together. We can set "name" to `null` if we want to inherit the device name for the entity. If we set an entity name the friendly_name wil be a combination of device and entity name. If "name" would be left away and a `device_class` is set, the entity name part will be derived from the `device_class`.
 
-To create a new sensor manually:
+- Example configuration payload with no name set and derived `device_class` name: `{"device_class": "motion", "state_topic": "homeassistant/binary_sensor/garden/state", "unique_id": "motion01ad", "device": {"identifiers": ["01ad"], "name": "Garden" }}`
+
+If no name is set, a default name will be set by MQTT ([see the MQTT platform documentation](#mqtt-discovery)).
+
+To create a new sensor manually and with the name set to `null` to derive device name "Garden":
```
Comment (jbouwh) [inline source/_integrations/mqtt.markdown]:
```suggestion
To create a new sensor manually and with the name set to `null` to derive the device name "Garden":
```

File: source/_integrations/mqtt.markdown [hunk 2]
```diff
@@ -596,15 +596,19 @@ A motion detection device which can be represented by a [binary sensor](/integra
...
+- Configuration payload with derived device name:  `{"name": null, "device_class": "motion", "state_topic": "homeassistant/binary_sensor/garden/state", "unique_id": "motion01ad", "device": {"identifiers": ["01ad"], "name": "Garden" }}`
 - Retain: The -r switch is added to retain the configuration topic in the broker. Without this, the sensor will not be available after Home Assistant restarts.
 
-It is also a good idea to add a `unique_id` to allow changes to the entity and a `device` mapping so we can group all sensors of a device together.
+It is also a good idea to add a `unique_id` to allow changes to the entity and a `device` mapping so we can group all sensors of a device together. We can set "name" to `null` if we want to inherit the device name for the entity. If we set an entity name the friendly_name wil be a combination of device and entity name. If "name" would be left away and a `device_class` is set, the entity name part will be derived from the `device_class`.
```
Comment (frenck) [inline source/_integrations/mqtt.markdown]:
```suggestion
It is also a good idea to add a `unique_id` to allow changes to the entity and a `device` mapping so we can group all sensors of a device together. We can set "name" to `null` if we want to inherit the device name for the entity. If we set an entity name the `friendly_name` wil be a combination of the device and entity name. If "name" is left away and a `device_class` is set, the entity name part will be derived from the `device_class`.
```

File: source/_integrations/mqtt.markdown [hunk 3]
```diff
@@ -891,25 +895,33 @@ definition as a template. Make sure you escape the template blocks as like
 in the example below. Home Assistant will convert the result to a string
 and will pass it to the MQTT publish service.
 
+The example below shows how to publish a temperature sensor 'Bathroom Temperature'.
+The `device_class` is set, so it is not needed to set the "name" option. The entity
+will inherit the name from the device_class set and also support translations.
```
Comment (frenck) [inline source/_integrations/mqtt.markdown]:
```suggestion
will inherit the name from the `device_class` set and also support translations.
```

**Comment by jbouwh** [comment]
@frenck may we we can merge this one?

File: source/_integrations/mqtt.markdown (hunk 2 above)
Comment (c0ffeeca7) [inline source/_integrations/mqtt.markdown]:
```suggestion
It is also a good idea to add a `unique_id` to allow changes to the entity and a `device` mapping so we can group all sensors of a device together. We can set "name" to `null` if we want to inherit the device name for the entity. If we set an entity name, the `friendly_name` will be a combination of device and entity name. If `name` is left away and a `device_class` is set, the entity name part will be derived from the `device_class`.
```

File: source/_integrations/mqtt.markdown [hunk 4]
```diff
@@ -596,15 +596,19 @@ A motion detection device which can be represented by a [binary sensor](/integra
...
+- Configuration payload with derived device name:  `{"name": null, "device_class": "motion", "state_topic": "homeassistant/binary_sensor/garden/state", "unique_id": "motion01ad", "device": {"identifiers": ["01ad"], "name": "Garden" }}`
 - Retain: The -r switch is added to retain the configuration topic in the broker. Without this, the sensor will not be available after Home Assistant restarts.
 
-It is also a good idea to add a `unique_id` to allow changes to the entity and a `device` mapping so we can group all sensors of a device together.
+It is also a good idea to add a `unique_id` to allow changes to the entity and a `device` mapping so we can group all sensors of a device together. We can set "name" to `null` if we want to inherit the device name for the entity. If we set an entity name the `friendly_name` wil be a combination of the device and entity name. If "name" is left away and a `device_class` is set, the entity name part will be derived from the `device_class`.
```
Comment (c0ffeeca7) [inline source/_integrations/mqtt.markdown]:
```suggestion
It is also a good idea to add a `unique_id` to allow changes to the entity and a `device` mapping so we can group all sensors of a device together. We can set "name" to `null` if we want to inherit the device name for the entity. If we set an entity name, the `friendly_name` will be a combination of the device and entity name. If `name` is left away and a `device_class` is set, the entity name part will be derived from the `device_class`.
```

**Review by c0ffeeca7 — APPROVED** [review]
LGTM, @jbouwh :+1:

**Comment by jbouwh** [comment]
@c0ffeeca7 saw your comments to late, I'll open a follow-up PR to address them. Thnx for reviewing.

**Comment by jbouwh** [comment]
Oh, I see now you committed your suggestions before I merged the PR. Thnx.

**Pull request merged**